### ISSUE TITLE
base class function in code block no longer linked if the code block contains a derived class definition calling the base class function in Doxygen 1.9.5 or later

### ISSUE THREAD

**Issue by scott-zhong** [doxygen/doxygen#10915]
**Describe the bug**
base class function in code block no longer linked if the code block contains a derived class definition calling the base class function in Doxygen 1.9.5 or later

```
/**
 * \code
 * class Derived : public Base {
 *   void test() {
 *     fun();
 *   }
 * };
 * \endcode
 */
class Base {
public:
  /**
   * virtual function
   */
  virtual void fun();
};
```

**Screenshots**
Doxygen 1.9.4:
![image](https://github.com/doxygen/doxygen/assets/32584126/7e10dc1a-40a7-4104-be47-20f35102f032)
`fun()` is a link.

Doxygen 1.9.5 - trunk (3192bd6992de727664f17107022aa36de3a0d9d3):
![image](https://github.com/doxygen/doxygen/assets/32584126/51f8eb6f-a738-4604-a937-a7e5276dd24c)
`fun()` is not a link

**To Reproduce**
[testcase.zip](https://github.com/user-attachments/files/15517843/testcase.zip)

**Expected behavior**
`fun()` is a link.

**Additional context**
Using git bisect shows the following commit as the cause

```
b290399fab92df45376103d7c094f053f24495fa is the first bad commit
commit b290399fab92df45376103d7c094f053f24495fa
Author: Dimitri van Heesch <doxygen@gmail.com>
Date:   Tue May 24 08:12:07 2022 +0200

    Refactoring: reimplement getDefs using symbol resolver

 src/classdef.cpp       |  44 ++--
 src/classdef.h         |   6 +-
 src/code.l             |   2 +-
 src/definition.cpp     |   1 +
 src/doxygen.cpp        |  40 ++--
 src/doxygen.h          |   9 +-
 src/symbolresolver.cpp | 568 +++++++++++++++++++++++++++++++++++++++++++------
 src/symbolresolver.h   |  11 +
 src/util.cpp           | 173 +++++++++++++--
 src/vhdldocgen.cpp     |  12 +-
 10 files changed, 741 insertions(+), 125 deletions(-)
```

**Comment by doxygen** [comment]
@scott-zhong Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by scott-zhong** [comment]
@doxygen I have verify the fix.  Thank you!!!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.12.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
